File: src/main/java/com/rewards_service/dto/AwardResponse.java
package com.rewards_service.dto;

import java.util.UUID;

public class AwardResponse {

    private UUID userId;
    private long pointsAwarded;
    private UUID transactionId;

    public AwardResponse(UUID userId, long pointsAwarded, UUID transactionId) {
        this.userId = userId;
        this.pointsAwarded = pointsAwarded;
        this.transactionId = transactionId;
    }

    // Getters
    public UUID getUserId() {
        return userId;
    }

    public long getPointsAwarded() {
        return pointsAwarded;
    }

    public UUID getTransactionId() {
        return transactionId;
    }
}

File: src/main/kotlin/com/rewards_service/dto/AwardResponse.kt
package com.rewards_service.dto

import java.util.UUID

class AwardResponse {

    var userId: UUID? = null
    var pointsAwarded: Long = 0L
    var transactionId: UUID? = null

    constructor(userId: UUID?, pointsAwarded: Long, transactionId: UUID?) {
        this.userId = userId
        this.pointsAwarded = pointsAwarded
        this.transactionId = transactionId
    }
}
